skippy-core/src/main/java/io/skippy/core/SkippyTestApi.java

```diff
@@ -110,15 +142,17 @@ public boolean testNeedsToBeExecuted(Class<?> test) {
      * @param testClass a test class
      */
     public void prepareExecFileGeneration(Class<?> testClass) {
-        Profiler.profile("SkippyTestApi#prepareCoverageDataCaptureFor", () -> {
+        Profiler.profile("SkippyTestApi#prepareExecFileGeneration", () -> {
             swallowJacocoExceptions(() -> {
                 IAgent agent = RT.getAgent();
+                // Is testClass a nested test? Yes: Store the execution data that was generated for the parent before resetting the agent.
+                addExecutionDataToParent(asList(agent.getExecutionData(false)));
```

Comment:
\[Hige nitpick] Maybe we could have `addExecutionDataToParent` accept a `Supplier<List<byte<[]>>` instead, so that we can lazily evaluate `agent.getExecutionData` only when `testClass` is a nested class.

Reply:
Great feedback. I will incorporate this change in my final PR.